### PR TITLE
Port to Swift 3

### PR DESCRIPTION
**Pull request by basine**

### CODE REVIEW

File: Source/Model/Message/ConversationMessage+Reaction.swift [hunk 1]
```diff
@@ -19,33 +19,37 @@
 import Foundation
 
 
+public extension UUID {
```
Comment (typfel) [inline Source/Model/Message/ConversationMessage+Reaction.swift]: Should we not move this extension to transport?

File: Source/Model/UserClient/UserClient.swift [hunk 2]
```diff
@@ -91,24 +107,24 @@ public class UserClient: ZMManagedObject, UserClientType {
         // Fetch fingerprint if not there yet (could remain nil after fetch)
         if let managedObjectContext = self.managedObjectContext,
             let _ = self.remoteIdentifier
-            where managedObjectContext.zm_isSyncContext && self.fingerprint == .None
+            , managedObjectContext.zm_isSyncContext && self.fingerprint == .none
         {
             self.fingerprint = self.fetchFingerprint()
```
Comment (daehn) [inline Source/Model/UserClient/UserClient.swift]: Bad bot

File: Source/Model/Message/ConversationMessage+Reaction.swift (hunk 1 above)
Comment (basine) [inline Source/Model/Message/ConversationMessage+Reaction.swift]: Yeah we should

File: Source/Model/Confirmation/ZMMessageConfirmation.swift [hunk 3]
```diff
@@ -21,43 +21,43 @@ import CoreData
...
         case .READ:
-            return .Read
+            return .read
         }
     }
 }
 
 @objc(ZMMessageConfirmation)
-public class ZMMessageConfirmation: NSManagedObject {
+open class ZMMessageConfirmation: NSManagedObject {
```
Comment (daehn) [inline Source/Model/Confirmation/ZMMessageConfirmation.swift]: I think this could stay `public` as we never subclass it

**Comment by daehn** [comment]
The `.travis.yml` should be updated to use the `xcode8` image and an iOS 10 Simulator.

File: Source/Model/Message/AssetCache.swift [hunk 4]
```diff
@@ -72,39 +72,39 @@ struct PINAssetCache : Cache {
...
     
-    func hasDataForKey(key: String) -> Bool {
+    func hasDataForKey(_ key: String) -> Bool {
         return self.assetData(key) != nil
     }
 }
 
 // MARK: - Image assets
-public class ImageAssetCache : NSObject {
+open class ImageAssetCache : NSObject {
```
Comment (typfel) [inline Source/Model/Message/AssetCache.swift]: This can stay public I think. Same thing goes for the methods in this class.

File: Source/Model/Message/ConversationMessage+Deletion.swift [hunk 5]
```diff
@@ -46,48 +44,50 @@ extension ZMMessage {
...
-        guard !isZombieObject, let sender = sender where sender.isSelfUser else { return }
+        guard !isZombieObject, let sender = sender , sender.isSelfUser else { return }
         guard let conversation = conversation else { return }
         
         // We insert a message of type `ZMMessageDelete` containing the nonce of the message that should be deleted
         let deletedMessage = ZMGenericMessage(deleteMessage: nonce.transportString(), nonce: NSUUID().transportString())
         
-        conversation.appendGenericMessage(deletedMessage, expires:false, hidden: true)
-        removeMessageClearingSender(true)
+        conversation.append(deletedMessage!, expires:false, hidden: true)
```
Comment (typfel) [inline Source/Model/Message/ConversationMessage+Deletion.swift]: Why is `deletedMessage` optional here. Maybe we can the `ZMGenericMessage` constructor.

File: Tests/Source/Model/Observer/ObjectObserverToken/GlobalConversationObserverTests.swift [hunk 6]
```diff
@@ -381,28 +381,29 @@ class GlobalConversationObserverTests : ZMBaseManagedObjectTest {
...
         self.uiMOC.saveOrRollback()
         
         XCTAssertEqual(pendingList.count, 1)
         XCTAssertEqual(normalList.count, 0)
 
         // when
-        conversation.connection.status = .Accepted
-        conversation.conversationType = .OneOnOne
+        conversation.connection!
+            .status = .accepted
```
Comment (daehn) [inline Tests/Source/Model/Observer/ObjectObserverToken/GlobalConversationObserverTests.swift]: The newline could be removed

File: Tests/Source/Model/Observer/ObjectObserverToken/MessageObserverTokenTests.swift [hunk 7]
```diff
@@ -154,58 +154,58 @@ class MessageObserverTokenTests : ZMBaseManagedObjectTest {
...
+        sender.remoteIdentifier = UUID.create()
+        let message = ZMMessage.insertNewObject(in: self.uiMOC)
         message.sender = sender
-        sender.smallProfileRemoteIdentifier = NSUUID.createUUID()
+        sender.smallProfileRemoteIdentifier = UUID.create()
         
         // when
         self.checkThatItNotifiesTheObserverOfAChange(message,
-            modifier: { $0.sender!.imageSmallProfileData = self.verySmallJPEGData()},
+            modifier: { $0.sender!.imageSmallProfileData = self.verySmallJPEGData()!},
```
Comment (daehn) [inline Tests/Source/Model/Observer/ObjectObserverToken/MessageObserverTokenTests.swift]: Maybe `verySmallJPEGData` could return an implicitly unwrapped optional or a non optional and assert internally if something goes wrong.

File: Source/Model/Message/ZMFileMetadata.swift [hunk 8]
```diff
@@ -127,16 +147,16 @@ extension ZMFileMetadata {
     
     public var filename : String {
         get {
-            return  fileURL.lastPathComponent ?? "unnamed"
+            return  fileURL.lastPathComponent
```
Comment (typfel) [inline Source/Model/Message/ZMFileMetadata.swift]: `lastPathComponent` is an empty string if it's missing. We should keep the behaviour of returning `"unnamed"` in that case.

File: Source/Model/User/ZMUser.m [hunk 9]
```diff
@@ -956,6 +956,12 @@ - (void)setImageData:(NSData *)imageData forFormat:(ZMImageFormat)format propert
 - (void)setImageData:(NSData *)imageData forKey:(NSString *)key format:(ZMImageFormat)format
 {
     [self willChangeValueForKey:key];
+    
+    if (imageData == nil) {
```
Comment (typfel) [inline Source/Model/User/ZMUser.m]: This is new behaviour. Does it fix a bug?

File: Source/Model/Message/ZMFileMetadata.swift (hunk 8 above)
Comment (basine) [inline Source/Model/Message/ZMFileMetadata.swift]: Hmm I didn't make that change. Can this be automatic?

Comment (basine) [inline Source/Model/Message/ZMFileMetadata.swift]: Ah lastPathComponent is not optional anymore!

File: Source/Model/User/ZMUser.m (hunk 9 above)
Comment (basine) [inline Source/Model/User/ZMUser.m]: It was broken after porting because the methods called from ObjC expected non-optional types. I checked in the PinCache project what happens when you call setValueForKey with nil and decided it was safer to use the removeValueForKey instead.

File: Source/Model/Confirmation/ZMMessageConfirmation.swift (hunk 3 above)
Comment (basine) [inline Source/Model/Confirmation/ZMMessageConfirmation.swift]: Yeah I did not review the public vs open changes. I assumed we can fix this later.

File: Source/Model/Message/AssetCache.swift (hunk 4 above)
Comment (basine) [inline Source/Model/Message/AssetCache.swift]: I did not review public vs open (since there are too many). I assumed we can do this later since we need to port zmessaging ASAP.

File: Source/Model/UserClient/UserClient.swift [hunk 10]
```diff
@@ -39,30 +39,46 @@ public let ZMUserClientRemoteIdentifierKey = "remoteIdentifier"
...
     @NSManaged public var messagesMissingRecipient: Set<ZMMessage>?
     @NSManaged public var numberOfKeysRemaining: Int32
     @NSManaged public var activationAddress: String?
-    @NSManaged public var activationDate: NSDate?
+    @NSManaged public var activationDate: Date?
     @NSManaged public var model: String?
     @NSManaged public var deviceClass: String?
-    @NSManaged public var activationLocationLatitude: Double
-    @NSManaged public var activationLocationLongitude: Double
+    @NSManaged public var activationLocationLatitude: NSNumber?
```
Comment (basine) [inline Source/Model/UserClient/UserClient.swift]: Because UserClient subclasses from ZMManagedObject and it was complaining that those non-optional values are not initialized when calling super.init. So I had to use a type that can be optional while still being visible from ObjC (Double? is not). I made some subsequent changes to the model though - so maybe it works now.

File: Tests/Source/Model/Observer/ObjectObserverToken/MessageObserverTokenTests.swift (hunk 7 above)
Comment (basine) [inline Tests/Source/Model/Observer/ObjectObserverToken/MessageObserverTokenTests.swift]: It's defined in ObjC in ZMTesting.

File: Cartfile.resolved [hunk 11]
```diff
@@ -3,11 +3,11 @@ github "wireapp/libPhoneNumber-iOS" "0.8.12"
...
 github "wireapp/ono" "1.2.3"
 github "wireapp/protobuf-objc" "1.9.11"
-github "wireapp/wire-ios-cryptobox" "3.1.4"
-github "wireapp/wire-ios-system" "14.40.4"
-github "wireapp/wire-ios-link-preview" "3.7.0"
-github "wireapp/wire-ios-utilities" "10.30.3"
-github "wireapp/wire-ios-images" "4.18.5"
-github "wireapp/wire-ios-testing" "6.31.6"
-github "wireapp/wire-ios-transport" "9.0.1"
+github "wireapp/wire-ios-cryptobox" "4.0.0"
```
Comment (daehn) [inline Cartfile.resolved]: These should also be `~>` pinned in the `Cartfile` to the new major version

File: Source/Model/Message/ZMFileMetadata.swift [hunk 12]
```diff
@@ -20,18 +20,38 @@
 import Foundation
 import MobileCoreServices
 import ZMCSystem
+fileprivate func < <T : Comparable>(lhs: T?, rhs: T?) -> Bool {
```
Comment (daehn) [inline Source/Model/Message/ZMFileMetadata.swift]: I can add it to `wire-ios-utilities` and make a new release

File: Source/Model/UserClient/UserClient.swift (hunk 10 above)
Comment (basine) [inline Source/Model/UserClient/UserClient.swift]: Nope, still does not work without these changes

Comment (typfel) [inline Source/Model/UserClient/UserClient.swift]: Can't we just remove the optional and initialise them to some default value?

File: Source/Model/Message/ZMFileMetadata.swift (hunk 12 above)
Comment (daehn) [inline Source/Model/Message/ZMFileMetadata.swift]: Done in https://github.com/wireapp/wire-ios-utilities/pull/8

File: Source/Model/UserClient/UserClient.swift (hunk 10 above)
Comment (basine) [inline Source/Model/UserClient/UserClient.swift]: No, if it's not optional it will complain - even if there is a default value defined in the data model. You can try it yourself ;)

File: Tests/Source/Model/ConversationList/ZMConversationListTests.m [hunk 13]
```diff
@@ -526,7 +526,7 @@ - (void)testThatAddingMessageToClearedConversationMovesItToActiveConversationsLi
     ZMConversation *c1 = [ZMConversation insertNewObjectInManagedObjectContext:self.uiMOC];
     c1.conversationType = ZMConversationTypeOneOnOne;
     c1.lastModifiedDate = [NSDate date];
-    ZMMessage *message = [c1 appendMessageWithText:@"message"];
+    ZMMessage *message = (id)[c1 appendMessageWithText:@"message"];
```
Comment (typfel) [inline Tests/Source/Model/ConversationList/ZMConversationListTests.m]: Which should change this into `ZMConversationMessage`. But I guess also at some point later.

File: Source/Model/Message/LinkPreview+ProtocolBuffer.swift [hunk 14]
```diff
@@ -26,8 +26,8 @@ extension LinkPreview {
         self.init(originalURLString: protocolBuffer.url, permamentURLString: protocolBuffer.permanentURLString, offset: Int(protocolBuffer.urlOffset))
     }
     
-    public var protocolBuffer: ZMLinkPreview {
-        let linkPreviewBuilder = ZMLinkPreview.builder()
+    public var protocolBuffer: ZMLinkPreview? {
+        let linkPreviewBuilder = ZMLinkPreview.builder()!
```
Comment (marcoconti83) [inline Source/Model/Message/LinkPreview+ProtocolBuffer.swift]: Is there a reason why we changed this var to optional, but then force unwrap the `.builder` method? To me either we do one or the other.
I think it's fair to crash if we can't assemble a protocol buffer anyway, maybe adding the `!` if needed (maybe even in the return line if needed) is enough.

File: Source/Model/Message/ZMClientMessage+Encryption.swift [hunk 15]
```diff
@@ -77,30 +74,33 @@ extension ZMGenericMessage {
...
     /// Returns a message with recipients and a strategy to handle missing clients
-    private func otrMessage(selfClient: UserClient,
+    fileprivate func otrMessage(_ selfClient: UserClient,
                             conversation: ZMConversation,
-                            externalData: NSData?,
+                            externalData: Data?,
                             sessionDirectory: EncryptionSessionsDirectory) -> (message: ZMNewOtrMessage, strategy: MissingClientsStrategy) {
         
-        let recipientUsers : [ZMUser]
+        var recipientUsers : [ZMUser] = []
```
Comment (marcoconti83) [inline Source/Model/Message/ZMClientMessage+Encryption.swift]: Did the behavior of uninitialized `let`s change in Swift 3? Previously I could leave this as a `let`, not initialize it here, and initialize it later in all code branches. Losing the `let` in favor of a `var` might cause unwanted overwrite to go unnoticed.

File: Source/Model/UserClient/UserClient.swift [hunk 16]
```diff
@@ -177,10 +193,10 @@ public class UserClient: ZMManagedObject, UserClientType {
     /// Can be called several times without issues
     public func resetSession() {
         // Delete should happen on sync context since the cryptobox could be accessed only from there
-        UserClient.deleteSession(forClientWithRemoteIdentifier: self.remoteIdentifier, managedObjectContext: (self.managedObjectContext?.zm_syncContext)!)
+        UserClient.deleteSession(forClientWithRemoteIdentifier: self.remoteIdentifier, managedObjectContext: (self.managedObjectContext?.zm_sync)!)
```
Comment (marcoconti83) [inline Source/Model/UserClient/UserClient.swift]: I would have preferred to keep `zm_syncContext` because `zm_sync` doesn't really say much (what is a "sync"?). I know that "context" is redundant as we already have the type to tell that this is a `NSManagedObject`, but `zm_sync` to me still sounds as it's missing something.

File: Tests/Source/Model/Messages/ZMClientMessageTests+OTR.swift [hunk 17]
```diff
@@ -186,25 +182,23 @@ extension ClientMessageTests_OTR {
...
             textMessage.senderClientID = senderID
             let confirmationMessage = textMessage.confirmReception()
             
             //when
-            guard let payloadAndStrategy = confirmationMessage.encryptedMessagePayloadData() else {
-                XCTFail()
-                return
-            }
+            guard let payloadAndStrategy = confirmationMessage?.encryptedMessagePayloadData()
+            else { return XCTFail()}
```
Comment (typfel) [inline Tests/Source/Model/Messages/ZMClientMessageTests+OTR.swift]: Does this return & fail or just return?

File: Source/Notifications/ObjectObserverTokens/ConversationObserverToken.swift [hunk 18]
```diff
@@ -74,14 +74,14 @@ public class GeneralConversationChangeInfo : ObjectChangeInfo {
         var dict : Dictionary<String, NSObject?> = [:]
         
         for key in keys {
-            dict[key] = ""
+            dict[key] = "" as NSObject??
```
Comment (marcoconti83) [inline Source/Notifications/ObjectObserverTokens/ConversationObserverToken.swift]: I'm confused, why double optional here?

File: Tests/Source/Model/Messages/ZMClientMessageTests+OTR.swift (hunk 17 above)
Comment (daehn) [inline Tests/Source/Model/Messages/ZMClientMessageTests+OTR.swift]: Returns and fails

File: Source/Notifications/ObjectObserverTokens/Helpers/SetSnapshot.swift [hunk 19]
```diff
@@ -69,34 +69,34 @@ struct SetStateUpdate {
 
  struct SetSnapshot {
     
-    let set : OrderedSet<NSObject>
+    let set : NSOrderedSet
     let moveType : ZMSetChangeMoveType
     
-    init(set: OrderedSet<NSObject>,  moveType : ZMSetChangeMoveType) {
-        self.set = set
+    init(set: NSOrderedSet,  moveType : ZMSetChangeMoveType) {
```
Comment (marcoconti83) [inline Source/Notifications/ObjectObserverTokens/Helpers/SetSnapshot.swift]: A very annoying person would point out that there are two spaces here :)

File: Source/Notifications/ZMNotifications.m [hunk 20]
```diff
@@ -167,7 +167,10 @@ @implementation ZMMessageNotification
 
 + (id<ZMMessageObserverOpaqueToken>)addMessageObserver:(id<ZMMessageObserver>)observer forMessage:(id<ZMConversationMessage>)conversationMessage;
 {
-    return (id) [[MessageObserverToken alloc] initWithObserver: observer object:conversationMessage];
+    if ([conversationMessage isKindOfClass:[ZMMessage class]]) {
+    return (id) [[MessageObserverToken alloc] initWithObserver: observer object:(ZMMessage *)conversationMessage];
```
Comment (marcoconti83) [inline Source/Notifications/ZMNotifications.m]: Indentation

File: Tests/Source/Model/Observer/ManagedObjectContextObserverTests.swift [hunk 21]
```diff
@@ -298,36 +298,36 @@ class ManagedObjectContextObserverTests : ZMBaseManagedObjectTest {
...
         self.uiMOC.globalManagedObjectContextObserver.isTesting = true
-        let filename = "foo.mp4"
-        let documents = NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true).first!
-        let documentsURL = NSURL(fileURLWithPath: documents)
-        let fileURL =  documentsURL.URLByAppendingPathComponent(filename)
-        verySmallJPEGData().writeToURL(fileURL, atomically: true)
-        defer { try! NSFileManager.defaultManager().removeItemAtURL(fileURL) }
-        
-        let conversation = ZMConversation.insertNewObjectInManagedObjectContext(self.uiMOC)
+        let filename = "video.mp4"
```
Comment (marcoconti83) [inline Tests/Source/Model/Observer/ManagedObjectContextObserverTests.swift]: "Foo" is not cool anymore? :) The change is OK, I was wondering if there is any reason why we changed it (or if it was just casual).

File: Tests/Source/Model/UserClient/UserClientKeyStoreTests.swift [hunk 22]
```diff
@@ -122,15 +120,15 @@ class UserClientKeysStoreTests: OtrBaseTest {
...
+        try! FileManager.default.createDirectory(atPath: UserClientKeysStore.legacyOtrDirectory.path, withIntermediateDirectories: true, attributes: [:])
     }
     
     func testThatTheNonEmptyLegacyOTRFolderIsDetected() {
         
         // given
-        self.dynamicType.createFakeOTRFolder()
-        "foo".dataUsingEncoding(NSUTF8StringEncoding)!.writeToURL(UserClientKeysStore.legacyOtrDirectory.URLByAppendingPathComponent("dummy.txt"), atomically: true)
+        type(of: self).createFakeOTRFolder()
+        try? "foo".data(using: String.Encoding.utf8)!.write(to: UserClientKeysStore.legacyOtrDirectory.appendingPathComponent("dummy.txt"), options: Data.WritingOptions.atomic)
```
Comment (marcoconti83) [inline Tests/Source/Model/UserClient/UserClientKeyStoreTests.swift]: Should we use `try!` instead? This way it will not silently fail.

File: Tests/Source/Model/UserClient/UserClientKeyStoreTests.swift [hunk 23]
```diff
@@ -170,17 +168,17 @@ class UserClientKeysStoreTests: OtrBaseTest {
     func testThatItMovesTheLegacyCryptobox() {
         
         // given
-        self.dynamicType.cleanOTRFolder()
+        type(of: self).cleanOTRFolder()
 
-        self.dynamicType.createFakeOTRFolder()
-        "foo".dataUsingEncoding(NSUTF8StringEncoding)!.writeToURL(UserClientKeysStore.legacyOtrDirectory.URLByAppendingPathComponent("dummy.txt"), atomically: true)
+        type(of: self).createFakeOTRFolder()
+        try? "foo".data(using: String.Encoding.utf8)!.write(to: UserClientKeysStore.legacyOtrDirectory.appendingPathComponent("dummy.txt"), options: Data.WritingOptions.atomic)
```
Comment (marcoconti83) [inline Tests/Source/Model/UserClient/UserClientKeyStoreTests.swift]: Should we use `try!` instead? This way it will not silently fail.

File: Tests/Source/Model/UserClient/UserClientTests.swift [hunk 24]
```diff
@@ -164,21 +158,20 @@ class UserClientTests: ZMBaseManagedObjectTest {
...
     func testThatItDeletesASessionWhenDeletingAClient() {
         self.syncMOC.performGroupedBlockAndWait{
             // given
-            let selfClient = self.createSelfClientOnMOC(self.syncMOC)
+            let selfClient = self.createSelfClient(onMOC: self.syncMOC)
             var preKeys : [(id: UInt16, prekey: String)] = []
             selfClient.keysStore.encryptionContext.perform({ (sessionsDirectory) in
-                preKeys = try! sessionsDirectory.generatePrekeys(Range(0..<2)).flatMap{$0}
-            })
+                preKeys = try! sessionsDirectory.generatePrekeys(CountableRange<UInt16>(0..<2))            })
```
Comment (marcoconti83) [inline Tests/Source/Model/UserClient/UserClientTests.swift]: Weird indent

File: Tests/Source/Model/UserClient/UserClientTests.swift [hunk 25]
```diff
@@ -272,34 +265,37 @@ class UserClientTests: ZMBaseManagedObjectTest {
...
             fetchRequest.fetchLimit = 1
             // when
-            let fetchedClient = self.syncMOC.executeFetchRequestOrAssert(fetchRequest).first as? UserClient
-            
-            // then
-            XCTAssertNotNil(fetchedClient)
-            XCTAssertNotNil(fetchedClient!.fingerprint)
+            var fetchedClient: UserClient?
+            do {
+                fetchedClient = try self.syncMOC.fetch(fetchRequest).first
```
Comment (marcoconti83) [inline Tests/Source/Model/UserClient/UserClientTests.swift]: The definition of `fetchedClient` can be moved inside the do and turned into a `let`

File: Source/Model/Message/ConversationMessage+Reaction.swift (hunk 1 above)
Comment (daehn) [inline Source/Model/Message/ConversationMessage+Reaction.swift]: I am on it

File: Source/Model/Message/ConversationMessage+Deletion.swift (hunk 5 above)
Comment (basine) [inline Source/Model/Message/ConversationMessage+Deletion.swift]: https://github.com/wireapp/wire-ios-protos/pull/10

File: Tests/Source/Model/Observer/ObjectObserverToken/MessageObserverTokenTests.swift (hunk 7 above)
Comment (basine) [inline Tests/Source/Model/Observer/ObjectObserverToken/MessageObserverTokenTests.swift]: https://github.com/wireapp/wire-ios-testing/pull/5

File: Source/Model/Message/ZMFileMetadata.swift (hunk 8 above)
Comment (basine) [inline Source/Model/Message/ZMFileMetadata.swift]: Fixed

File: Cartfile.resolved (hunk 11 above)
Comment (basine) [inline Cartfile.resolved]: Fixed

File: Source/Model/Message/ZMClientMessage+Encryption.swift (hunk 15 above)
Comment (basine) [inline Source/Model/Message/ZMClientMessage+Encryption.swift]: I guess that was an automatic change.

File: Source/Model/UserClient/UserClient.swift (hunk 16 above)
Comment (basine) [inline Source/Model/UserClient/UserClient.swift]: Bridging from ObjC unfortunately drops the `Context` in Swift

File: Source/Notifications/ObjectObserverTokens/ConversationObserverToken.swift (hunk 18 above)
Comment (basine) [inline Source/Notifications/ObjectObserverTokens/ConversationObserverToken.swift]: That was an automatic change

File: Source/Notifications/ObjectObserverTokens/Helpers/SetSnapshot.swift (hunk 19 above)
Comment (basine) [inline Source/Notifications/ObjectObserverTokens/Helpers/SetSnapshot.swift]: Which were there before ;)

File: Tests/Source/Model/Observer/ManagedObjectContextObserverTests.swift (hunk 21 above)
Comment (basine) [inline Tests/Source/Model/Observer/ManagedObjectContextObserverTests.swift]: I thought the test broke because of the missing file. So I added a file with that name - turned out something else was broken. Didn't revert the change.

File: Tests/Source/Model/UserClient/UserClientKeyStoreTests.swift (hunk 22 above)
Comment (basine) [inline Tests/Source/Model/UserClient/UserClientKeyStoreTests.swift]: maybe a if (try? ...) instead

File: Source/Model/Message/LinkPreview+ProtocolBuffer.swift (hunk 14 above)
Comment (basine) [inline Source/Model/Message/LinkPreview+ProtocolBuffer.swift]: My guess is that it was an automatic change.

File: Source/Model/Message/ConversationMessage+Reaction.swift (hunk 1 above)
Comment (daehn) [inline Source/Model/Message/ConversationMessage+Reaction.swift]: Done in https://github.com/wireapp/wire-ios-transport/pull/7

File: Tests/Source/Model/Observer/ObjectObserverToken/GlobalConversationObserverTests.swift (hunk 6 above)
Comment (basine) [inline Tests/Source/Model/Observer/ObjectObserverToken/GlobalConversationObserverTests.swift]: Fixed

File: Source/Model/Message/ZMClientMessage+Encryption.swift (hunk 15 above)
Comment (basine) [inline Source/Model/Message/ZMClientMessage+Encryption.swift]: I guess that's how you do it now - no more declaring a let that is set afterwards.

Comment (daehn) [inline Source/Model/Message/ZMClientMessage+Encryption.swift]: If all branches initialize it it should still be possible

Comment (daehn) [inline Source/Model/Message/ZMClientMessage+Encryption.swift]: and here there is one branch where it would not be set now (if the `fetch` call returns `nil`)

Comment (marcoconti83) [inline Source/Model/Message/ZMClientMessage+Encryption.swift]: It should work if you add:
```
} else {
    recipientUsers = nil
}
```

after
```
if let message = ZMMessage.fetch(withNonce: messageID, for: conversation, in: conversation.managedObjectContext) {
     recipientUsers = [message.sender].flatMap { $0 }
}
```

**Comment by basine** [comment]
I updated the dependencies ZMUtilities, ZMTransport, ZMProtos and ZMTesting and made the changes in ZMDataModel

**Comment by daehn** [comment]
🙌
